pages/docs/reference/scope-functions.md

```diff
@@ -0,0 +1,530 @@
+    adam.apply {
+        age = 32
+        city = "London"
+    }.moveTo("Manchester")
+//sampleEnd
+}
+```
+</div>
+
+The lambda result is generally better for other purposes: assigning it to a variable, chaining operations on the result, and so on.
+
+<div class="sample" markdown="1" theme="idea">
+```kotlin
+fun main() {
+//sampleStart
+    val numbers = mutableListOf("one", "two", "three")
+    val resultSize = numbers.run { 
+        add("four")
+        add("five")
+        filter { it.endsWith("e") }.size        
+    }
+    println("There are $resultSize elements that end with e.")
+//sampleEnd
+}
+```
+</div>
+
+Additionally, you can ignore the return value and use a scope function for creating a temporary scope. 
+
+<div class="sample" markdown="1" theme="idea">
+```kotlin
+data class Person(var name: String, var age: Int, var city: String)
+
+fun main() {
+//sampleStart
+    val adam = Person("Adam", 20, "London")
+    val length = run {
+        val hello = "Hello"
+        val greeting = "$hello, ${adam.name}"
+        val length = greeting.length             // the variables are available only within the lambda
+        length
+    }
+//sampleEnd
+}
+```
+</div>
+
+## Functions
+
+To help you choose the right scope function for your case, we'll describe them in detail and provide usage recommendations. Technically, functions are interchangeable in many cases, so the examples show the conventions that define the common usage style. 
+
+### `let`
+
+**The context object** is available as an argument (`it`). **The return value** is the lambda result.
+
+`let` can be used to invoke one or more functions on results of call chains. For example, the following code prints the results of two operations on a collection:
+
+<div class="sample" markdown="1" theme="idea">
+```kotlin
+fun main() {
+//sampleStart
+    val numbers = mutableListOf("one", "two", "three", "four", "five")
+    val resultList = numbers.map { it.length }.filter { it > 3 }
+    println(resultList)    
+//sampleEnd
+}
+```
+</div>
+
+With `let`, you can rewrite it:
+
+<div class="sample" markdown="1" theme="idea">
+```kotlin
+fun main() {
+//sampleStart
+    val numbers = mutableListOf("one", "two", "three", "four", "five")
+    numbers.map { it.length }.filter { it > 3 }.let { 
+        println(it)
+        // and more function calls if needed
+    } 
+//sampleEnd
+}
+```
+</div>
+
+If the code block contains a single function with `it` as an argument, you can use the method reference (`::`) instead of the lambda:
+
+<div class="sample" markdown="1" theme="idea">
+```kotlin
+fun main() {
+//sampleStart
+    val numbers = mutableListOf("one", "two", "three", "four", "five")
+    numbers.map { it.length }.filter { it > 3 }.let(::println)
+//sampleEnd
+}
+```
+</div>
+
+`let` is often used for executing a code block only with non-null values. To perform actions on a non-null object, use the safe call operator `?` on it and call `let` with the actions in its lambda.
```

Comment:
This is actually the most common use of `let`, so it should be on top.

Reply:
I believe it became the most common even without us writing this page, so it won't harm to leave the narrative order going from basic usages to advanced. Otherwise `let` might get a reputation of a fancy `if (it != null)` replacement.